worlds/witness/__init__.py

```diff
@@ -0,0 +1,167 @@
+        self.items.adjust_after_options(self.locat, self.world, self.player)
+        self.regio = WitnessRegions(self.logic, self.locat)
+
+    def generate_basic(self):
+        # Generate item pool
+        pool = []
+        items_by_name = dict()
+        for item in self.items.ITEM_TABLE:
+            witness_item = self.create_item(item)
+            if item not in self.items.EVENT_ITEM_TABLE:
+                pool.append(witness_item)
+                items_by_name[item] = witness_item
+
+        # Put good item on first check
+        random_good_item = self.world.random.choice(self.items.GOOD_ITEMS)
+        first_check = self.world.get_location(
+            "Tutorial Gate Open", self.player
+        )
+        first_check.place_locked_item(items_by_name[random_good_item])
+        pool.remove(items_by_name[random_good_item])
+
+        # Put in junk items to fill the rest
+        junk_pool = self.items.JUNK_WEIGHTS.copy()
+        junk_pool = self.world.random.choices(
+            list(junk_pool.keys()), weights=list(junk_pool.values()),
+            k=len(self.locat.CHECK_LOCATION_TABLE) - len(pool)
+            - len(self.locat.EVENT_LOCATION_TABLE) - 1
+        )
+
+        pool += [self.create_item(junk) for junk in junk_pool]
+
+        # Tie Event Items to Event Locations (e.g. Laser Activations)
+        for event_location in self.locat.EVENT_LOCATION_TABLE:
+            item_obj = self.create_item(
+                self.logic.EVENT_ITEM_PAIRS[event_location]
+            )
+            location_obj = self.world.get_location(event_location, self.player)
+            location_obj.place_locked_item(item_obj)
+
+        self.world.itempool += pool
+
+    def create_regions(self):
+        self.regio.create_regions(self.world, self.player)
+
+    def set_rules(self):
+        set_rules(self.world, self.player, self.logic, self.locat)
+
+    def fill_slot_data(self) -> dict:
+        slot_data = self._get_slot_data()
+
+        for option_name in the_witness_options:
+            slot_data[option_name] = is_option_enabled(
+                self.world, self.player, option_name
+            )
+
+        slot_data["hard_mode"] = False
+
+        return slot_data
+
+    def create_item(self, name: str) -> Item:
+        item = self.items.ITEM_TABLE[name]
+        new_item = WitnessItem(
+            name, item.progression, item.code, player=self.player
+        )
+        new_item.trap = item.trap
+        return new_item
+
+    def get_filler_item_name(self) -> str: # Used ny itemlinks
+        junk_pool = self.items.JUNK_WEIGHTS.copy()
+
+        return self.world.random.choices(list(junk_pool.keys()), weights=list(junk_pool.values()))[0]
+
+class WitnessLocation(Location):
+    """
+    Archipelago Location for The Witness
+    """
+    game: str = "The Witness"
+    check_hex: int = -1
+
+    def __init__(self, player: int, name: str, address: typing.Optional[int],
+                 parent, ch_hex: int = -1):
```

Comment:
we do 120 chars per line in AP. no need to add a line break here
```suggestion
    def __init__(self, player: int, name: str, address: typing.Optional[int], parent, ch_hex: int = -1):
```

Reply:
Removed unnecessary line break